runtime/shibuya/src/xcm_config.rs

```diff
@@ -262,3 +385,312 @@ impl cumulus_pallet_dmp_queue::Config for Runtime {
+            match asset_id {
+                RELAY_ASSET => Some(PARENT),
+                20 => Some(PARACHAIN),
+                30 => Some(GENERAL_INDEX),
+                _ => None,
+            }
+        }
+
+        fn get_asset_id(asset_location: MultiLocation) -> Option<AssetId> {
+            match asset_location {
+                a if a == PARENT => Some(RELAY_ASSET),
+                a if a == PARACHAIN => Some(20),
+                a if a == GENERAL_INDEX => Some(30),
+                _ => None,
+            }
+        }
+    }
+
+    /// Helper struct used for testing `FixedRateOfForeignAsset`
+    struct ExecutionPayment;
+    impl ExecutionPaymentRate for ExecutionPayment {
+        fn get_units_per_second(asset_location: MultiLocation) -> Option<u128> {
+            match asset_location {
+                a if a == PARENT => Some(1_000_000),
+                a if a == PARACHAIN => Some(2_000_000),
+                a if a == GENERAL_INDEX => Some(3_000_000),
+                _ => None,
+            }
+        }
+    }
+
+    /// Execution fee for the specified weight, using provided `units_per_second`
+    fn execution_fee(weight: Weight, units_per_second: u128) -> u128 {
+        units_per_second * (weight as u128) / (WEIGHT_PER_SECOND as u128)
+    }
+
+    #[test]
+    fn asset_location_to_id() {
+        // Test cases where the MultiLocation is valid
+        assert_eq!(
+            AssetLocationIdConverter::<AssetId, AssetLocationMapper>::convert_ref(PARENT),
+            Ok(u128::max_value())
+        );
+        assert_eq!(
+            AssetLocationIdConverter::<AssetId, AssetLocationMapper>::convert_ref(PARACHAIN),
+            Ok(20)
+        );
+        assert_eq!(
+            AssetLocationIdConverter::<AssetId, AssetLocationMapper>::convert_ref(GENERAL_INDEX),
+            Ok(30)
+        );
+
+        // Test case where MultiLocation isn't supported
+        assert_eq!(
+            AssetLocationIdConverter::<AssetId, AssetLocationMapper>::convert_ref(
+                MultiLocation::here()
+            ),
+            Err(())
+        );
+    }
+
+    #[test]
+    fn asset_id_to_location() {
+        // Test cases where the AssetId is valid
+        assert_eq!(
+            AssetLocationIdConverter::<AssetId, AssetLocationMapper>::reverse_ref(u128::max_value()),
+            Ok(PARENT)
+        );
+        assert_eq!(
+            AssetLocationIdConverter::<AssetId, AssetLocationMapper>::reverse_ref(20),
+            Ok(PARACHAIN)
+        );
+        assert_eq!(
+            AssetLocationIdConverter::<AssetId, AssetLocationMapper>::reverse_ref(30),
+            Ok(GENERAL_INDEX)
+        );
+
+        // Test case where the AssetId isn't supported
+        assert_eq!(
+            AssetLocationIdConverter::<AssetId, AssetLocationMapper>::reverse_ref(0),
+            Err(())
+        );
+    }
+
+    // TODO: look into `TakeRevenue` implementation
```

Comment:
@akru I noticed we use this:

```
/// Null implementation just burns the revenue.
impl TakeRevenue for () {
	fn take_revenue(_revenue: MultiAsset) {}
}
```
Is this intentional or just a temporary solution?

Should we create a backlog item to add support for storing the revenue into some account?